examples/generate.sh

```diff
@@ -0,0 +1,106 @@
+#!/bin/bash
+# Copyright 2018 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+
+# Directories.
+SOURCE_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )"
+OUTPUT_DIR=${OUTPUT_DIR:-${SOURCE_DIR}/_out}
+ENVSUBST=${ENVSUBST:-envsubst}
+CLUSTERAWSADM=${CLUSTERAWSADM:-clusterawsadm}
+
+# Cluster name.
+export CLUSTER_NAME="${CLUSTER_NAME:-test1}"
+
+# Machine settings.
+export CONTROL_PLANE_MACHINE_TYPE="${CONTROL_PLANE_MACHINE_TYPE:-t2.medium}"
+export NODE_MACHINE_TYPE="${CONTROL_PLANE_MACHINE_TYPE:-t2.medium}"
+export SSH_KEY_NAME="${SSH_KEY_NAME:-default}"
+
+# Outputs.
+COMPONENTS_CLUSTER_API_GENERATED_FILE=${SOURCE_DIR}/provider-components/provider-components-cluster-api.yaml
+COMPONENTS_KUBEADM_GENERATED_FILE=${SOURCE_DIR}/provider-components/provider-components-kubeadm.yaml
+COMPONENTS_AWS_GENERATED_FILE=${SOURCE_DIR}/provider-components/provider-components-aws.yaml
+
+PROVIDER_COMPONENTS_GENERATED_FILE=${OUTPUT_DIR}/provider-components.yaml
+CLUSTER_GENERATED_FILE=${OUTPUT_DIR}/cluster.yaml
+CONTROLPLANE_GENERATED_FILE=${OUTPUT_DIR}/controlplane.yaml
+
+# Overwrite flag.
+OVERWRITE=0
+
+SCRIPT=$(basename "$0")
+while test $# -gt 0; do
+        case "$1" in
+          -h|--help)
+            echo "$SCRIPT - generates input yaml files for Cluster API on aws"
+            echo " "
+            echo "$SCRIPT [options]"
+            echo " "
+            echo "options:"
+            echo "-h, --help                show brief help"
+            echo "-f, --force-overwrite     if file to be generated already exists, force script to overwrite it"
+            exit 0
+            ;;
+          -f)
+            OVERWRITE=1
+            shift
+            ;;
+          --force-overwrite)
+            OVERWRITE=1
+            shift
+            ;;
+          *)
+            break
+            ;;
+        esac
+done
+
+if [ $OVERWRITE -ne 1 ] && [ -d "$OUTPUT_DIR" ]; then
+  echo "ERR: Folder ${OUTPUT_DIR} already exists. Delete it manually before running this script."
+  exit 1
+fi
+
+mkdir -p "${OUTPUT_DIR}"
+
+# Generate cluster resources.
+kustomize build "${SOURCE_DIR}/cluster" | envsubst > "${CLUSTER_GENERATED_FILE}"
```

Comment:
should this use `kubectl kustomize` instead?

Reply:
It fails for me

```
Error: rawResources failed to read Resources: Load from path ../crd failed: '../crd' must be a file (got d='/Users/vince/go/src/sigs.k8s.io/cluster-api-provider-aws/config/crd')
```

Reply:
Do we need to enforce a minimum version of kustomize then?

Reply:
Yeah, I've been using v3.1, I can add a check in this script

Reply:
Added the apiVersion and Kind to the kustomize scripts, that should fix the version

Reply:
I'd rather use `kubectl kustomize` if possible. It sounds like we may need to be more creative with bases vs resources. (I am nowhere near a kustomize expert.)

Reply:
`kubectl kustomize` is a few versions behind and it's impossible right now to understand which version it's actually vendoring (at least I couldn't find it). For now, let's use plain `kustomize` and move to kubectl kustomize if/when it checks all the boxes.

Reply:
It's v2.0.3 in k/k master now (https://github.com/kubernetes/kubernetes/blob/d951d19ec4d8ec829e83d146200ea91b664b6fa6/go.mod#L175)

Reply:
Yeah, we're using v3.1 within Cluster API repositories